oshi-core/src/main/java/oshi/software/common/AbstractOperatingSystem.java

```diff
@@ -150,6 +150,9 @@ public String getManufacturer() {
                 throw new IllegalArgumentException("Unimplemented enum type: " + sort.toString());
             }
         }
+        if (limit == 0)

```

Comment:
Good idea here to avoid copying the same list we'll return, but you also want to do this in the case that the list size is less than or equal to the limit, so you might consider moving this down and integrating with the other list logic.   (Also we should consider using a sublist rather than a new list and a bunch of adds.)